File: test_java/AutomatonTest.java
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class AutomatonTest {

	@Test
	void testConstructor() {
		Automaton eca = new Automaton(22, new Generation(false, true, false));
		assertEquals(22, eca.getRuleNum());
		assertArrayEquals(new boolean[] { false, true, false }, 
				eca.getGeneration(0).getStates());
		assertEquals(0, eca.getTotalSteps());
		assertEquals("010", eca.toString());
	}

	@Test
	void testEvolve() {
		Automaton eca = new Automaton(22, 
				new Generation("000000010000000", '1'));
		assertEquals(0, eca.evolve(-1));

		assertEquals(1, eca.evolve(1));
		assertEquals(1, eca.getTotalSteps());
		String evolution = 
				"000000010000000" + System.lineSeparator() + 
				"000000111000000";
		assertEquals(evolution, eca.toString());

		assertEquals(2, eca.evolve(2));
		assertEquals(3, eca.getTotalSteps());
		eca.falseSymbol = '_';
		eca.trueSymbol = '%';
		evolution = 
				"_______%_______" + System.lineSeparator() + 
				"______%%%______" + System.lineSeparator() + 
				"_____%___%_____" + System.lineSeparator() + 
				"____%%%_%%%____";
		assertEquals(evolution, eca.toString());

		Generation gen = eca.getGeneration(7);
		assertEquals("OOO.OOO.OOO.OOO", gen.getStates('.', 'O'));
		assertEquals(7, eca.getTotalSteps());
		eca.falseSymbol = '.';
		eca.trueSymbol = 'O';
		evolution = 
				".......O......." + System.lineSeparator() + 
				"......OOO......" + System.lineSeparator() + 
				".....O...O....." + System.lineSeparator() + 
				"....OOO.OOO...." + System.lineSeparator() + 
				"...O.......O..." + System.lineSeparator() + 
				"..OOO.....OOO.." + System.lineSeparator() + 
				".O...O...O...O." + System.lineSeparator() + 
				"OOO.OOO.OOO.OOO";
		assertEquals(evolution, eca.toString());
	}

	@Test
	void testGetCurrentGenerationZero() {
		Automaton eca = new Automaton(22, 
				new Generation("000000010000000", '1'));
		assertEquals("000000010000000", 
				eca.getCurrentGeneration().getStates(eca.falseSymbol, eca.trueSymbol));
	}

	@Test
	void testGetCurrentGenerationFive() {
		Automaton eca = new Automaton(22, 
				new Generation("000000010000000", '1'));
		eca.evolve(5);
		assertEquals("001110000011100", 
				eca.getCurrentGeneration().getStates(eca.falseSymbol, eca.trueSymbol));
	}
}

File: MainProj1_DataAbsProj1/MainProjectTests/AutomatonTest.cs
using FluentAssertions;
using MainProject;
using Xunit;

namespace MainProjectTests
{
    public class AutomatonTest
    {
        [Fact]
        public void ShouldConstruct()
        {
            Automaton eca = new Automaton(22, new Generation(false, true, false));
            eca.Should().Be(eca.GetRuleNum());
            eca.getGeneration(0).getStates().Should().BeEquivalentTo(new bool[] { false, true, false });
            eca.GetTotalSteps().Should().Be(0);
            eca.ToString().Should().Be("010");
        }

        [Fact]
        public void ShouldEvolve()
        {
            Automaton eca = new Automaton(22, new Generation("000000010000000", '1'));
            eca.Evolve(-1).Should().Be(0);
            
            eca.Evolve(1).Should().Be(1);
            eca.GetTotalSteps().Be(1);
            
            string evolution =  
            "000000010000000" + Environment.NewLine + 
            "000000111000000";
            eca.ToString().Should().Be(evolution);
            eca.Evolve(2).Should().Be(2);
            eca.GetTotalSteps().Should().Be(3);
            eca.FalseSymbol = '_';
            eca.TrueSymbol = '%';
            evolution = 
				"_______%_______" + Environment.NewLine + 
				"______%%%______" + Environment.NewLine + 
				"_____%___%_____" + Environment.NewLine + 
				"____%%%_%%%____";
            eca.ToString().Should().Be(evolution);
            
            Generation gen = eca.getGeneration(7);
            gen.GetStates('.', 'O').Should().Be("OOO.OOO.OOO.OOO");
            eca.GetTotalSteps().Should().Be(7);
            eca.FalseSymbol = '.';
            eca.TrueSymbol = 'O';
            evolution = 
				".......O......." + Environment.NewLine + 
				"......OOO......" + Environment.NewLine + 
				".....O...O....." + Environment.NewLine + 
				"....OOO.OOO...." + Environment.NewLine + 
				"...O.......O..." + Environment.NewLine + 
				"..OOO.....OOO.." + Environment.NewLine + 
				".O...O...O...O." + Environment.NewLine + 
				"OOO.OOO.OOO.OOO";
            eca.ToString().Should().Be(evolution);
        }

        [Fact]
        public void ShouldGetCurrentZero()
        {
            Automaton eca = new Automaton(22, 
				new Generation("000000010000000", '1'));
            eca.GetCurrentGeneration().GetStates(eca.falseSymbol, eca.trueSymbol).Should().Be("000000010000000");
        }

        [Fact]
        public void ShouldGetCurrentGenerationFive()
        {
            Automaton eca = new Automaton(22, 
				new Generation("000000010000000", '1'));
            eca.Evolve(5);
            eca.GetCurrentGeneration().GetStates(eca.falseSymbol, eca.trueSymbol).Should().Be("001110000011100");
        }

    }
}